terraform/main.tf

```diff
@@ -1,7 +1,7 @@
 data "aws_caller_identity" "current" {}
 
-locals  {
-    prefix = "${var.app}-${var.stage}"
-    attachments_bucket = "${terraform.workspace}-${var.app}-${var.stage}-attachments.${var.domain}"
-    emails_bucket = "${terraform.workspace}-${var.app}-${var.stage}-emails.${var.domain}"
+locals {
+  prefix             = "${var.app}-${var.stage}"
+  attachments_bucket = "${terraform.workspace}-${local.prefix}-attachments.${var.domain}"
```

Comment:
I was trying to do that before, but it wouldn't pick local.prefix and would return empty. Did you test that it gets you app and stage?

Reply:
Yeah, it created the user (one, possibly only thing that still uses `local.prefix`) as emails-dev-nifi.
Trying to think how you would have seen the behaviour you saw before. var.app has a default, so you can't miss that off, and stage doesn't, but the pipeline adds that in as TF_VARS_stage environment variable. Odd.